### PR TITLE
[WIP] fix static analized high priority issues

### PR DESCRIPTION
**Pull request by maxim-s**
all high priority issues here https://gist.github.com/maxim-s/41ef6139bebe0e728594f5ded28f2b58

### CODE REVIEW

File: src/core/Akka.Remote.TestKit/DataTypes.cs [hunk 1]
```diff
@@ -434,7 +434,7 @@ public override int GetHashCode()
             {
                 int hashCode = (_node != null ? _node.GetHashCode() : 0);
                 hashCode = (hashCode * 397) ^ (_target != null ? _target.GetHashCode() : 0);
-                hashCode = (hashCode * 397) ^ (_direction != null ? _direction.GetHashCode() : 0);
```
Comment (maxim-s) [inline src/core/Akka.Remote.TestKit/DataTypes.cs]: enum can't be null

File: src/core/Akka.Remote/EndpointManager.cs [hunk 2]
```diff
@@ -730,7 +730,6 @@ protected void Accepting()
                         _endpoints.MarkAsQuarantined(gotuid.RemoteAddress, gotuid.Uid,
                             Deadline.Now + _settings.QuarantineDuration);
                         _eventPublisher.NotifyListeners(new QuarantinedEvent(gotuid.RemoteAddress, gotuid.Uid));
-                        Context.Stop(pass.Endpoint);
```
Comment (maxim-s) [inline src/core/Akka.Remote/EndpointManager.cs]: if policy is WasGated Context.Stop(pass.Endpoint) will cause null reference exception
https://github.com/akka/akka/blob/master/akka-remote/src/main/scala/akka/remote/Remoting.scala#L704

File: src/core/Akka/Actor/ActorCell.FaultHandling.cs [hunk 3]
```diff
@@ -59,31 +59,29 @@ private void FaultRecreate(Exception cause)
                 if (System.Settings.DebugLifecycle)
                     Publish(new Debug(_self.Path.ToString(), failedActor.GetType(), "Restarting"));
 
-                if (failedActor != null)
```
Comment (maxim-s) [inline src/core/Akka/Actor/ActorCell.FaultHandling.cs]: failedActor is _actor, _actor already checked on null

File: src/core/Akka/Actor/Futures.cs [hunk 4]
```diff
@@ -40,18 +40,9 @@ public static Task<T> Ask<T>(this ICanTell self, object message, TimeSpan? timeo
             if (provider == null)
                 throw new ArgumentException("Unable to resolve the target Provider", nameof(self));
 
-            ResolveReplyTo();
```
Comment (maxim-s) [inline src/core/Akka/Actor/Futures.cs]: the same as do nothing

**Comment by Danthar** [comment]
Neat. Nice work @maxim-s

File: src/core/Akka.Remote/EndpointManager.cs (hunk 2 above)
Comment (Aaronontheweb) [inline src/core/Akka.Remote/EndpointManager.cs]: NICE catch.

**Pull request merged**